packages/a11y-base/test/aria-hidden.test.js

```diff
@@ -0,0 +1,212 @@
+
+        unhide();
+
+        // Elements are no longer hidden
+        expect(target2.parentNode.hasAttribute(attr)).to.be.false;
+        expect(sibling.hasAttribute(attr)).to.be.false;
+
+        // Hidden elements are still hidden
+        expect(hidden1.getAttribute(attr)).to.equal(attr === 'inert' ? null : 'true');
+        expect(hidden2.getAttribute(attr)).to.equal(attr === 'inert' ? null : '');
+
+        // Marker attributes are removed
+        expect(hidden1.hasAttribute(`data-${attr}`)).to.be.false;
+        expect(hidden2.hasAttribute(`data-${attr}`)).to.be.false;
+      });
+
+      it(`should set ${attr} on other nodes when passing multiple targets`, () => {
+        const unhide = hideFunc([target1, target2], parent);
+        callbacks.add(unhide);
+
+        // First target node is not hidden
+        expect(target1.hasAttribute(attr)).to.be.false;
+
+        // Second target node is not hidden
+        expect(target2.hasAttribute(attr)).to.be.false;
+
+        // Parent node is not hidden either
+        expect(target2.parentNode.hasAttribute(attr)).to.be.false;
+
+        // Sibling element is hidden
+        expect(sibling.getAttribute(attr)).to.equal('true');
+      });
+
+      it(`should remove ${attr} from other nodes when passing multiple targets`, () => {
+        const unhide = hideFunc([target1, target2], parent);
+        unhide();
+
+        // Sibling element is no longer hidden
+        expect(sibling.hasAttribute(attr)).to.be.false;
+
+        // Hidden elements are still hidden
+        expect(hidden1.getAttribute(attr)).to.equal(attr === 'inert' ? null : 'true');
+        expect(hidden2.getAttribute(attr)).to.equal(attr === 'inert' ? null : '');
+      });
+
+      it(`should set ${attr} correctly when calling on two different targets`, () => {
+        const unhide1 = hideFunc(target1, parent);
+        callbacks.add(unhide1);
+
+        // Everything but the target node is hidden
+        expect(target1.hasAttribute(attr)).to.be.false;
+        expect(target2.parentNode.getAttribute(attr)).to.equal('true');
+        expect(sibling.getAttribute(attr)).to.equal('true');
+
+        const unhide2 = hideFunc(target2, parent);
+        callbacks.add(unhide2);
+
+        // And now the first target is also hidden
+        expect(target1.getAttribute(attr)).to.equal('true');
+        expect(target2.parentNode.getAttribute(attr)).to.equal('true');
+        expect(sibling.getAttribute(attr)).to.equal('true');
+      });
+
+      it(`should remove ${attr} correctly when calling on two different targets`, () => {
+        const unhide1 = hideFunc(target1, parent);
+        const unhide2 = hideFunc(target2, parent);
+
+        unhide1();
+        unhide2();
+
+        // All the elements are no longer hidden
+        expect(target1.hasAttribute(attr)).to.be.false;
+        expect(target2.parentNode.hasAttribute(attr)).to.be.false;
+        expect(sibling.hasAttribute(attr)).to.be.false;
+      });
```

Comment:
I suggest adding a similar test that calls `unhide1()` only and verifies that the attributes are not removed on elements that still should be hidden by `target2`.

Reply:
Thanks, added a test to call `unhide2()` and verify that `target1` is no longer hidden.